File: src/main/java/com/elf/vipForElf/domain/floor/service/FloorService.java
package com.elf.vipForElf.domain.floor.service;

import com.elf.vipForElf.domain.floor.entity.FloorEntity;
import com.elf.vipForElf.domain.floor.vo.FloorInfoVO;
import com.elf.vipForElf.domain.floor.vo.FloorListVO;
import com.elf.vipForElf.web.dto.NewFloorDTO;

import java.util.List;

public interface FloorService {
    FloorInfoVO createFloor(NewFloorDTO newFloorDTO) throws IllegalAccessException;
    FloorInfoVO getFloorById(Long id);
    List<FloorListVO> getFloorList(String searchCondition, Integer page, Integer size);
    FloorEntity getFloorEntityById(Long id);
}

File: src/main/java/com/elf/vipForElf/domain/floor/service/FloorService.kt
package com.elf.vipForElf.domain.floor.service

import com.elf.vipForElf.domain.floor.entity.FloorEntity
import com.elf.vipForElf.domain.floor.vo.FloorInfoVO
import com.elf.vipForElf.domain.floor.vo.FloorListVO
import com.elf.vipForElf.web.dto.NewFloorDTO

interface FloorService {
    @Throws(IllegalAccessException::class)
    fun createFloor(newFloorDTO: NewFloorDTO): FloorInfoVO

    fun getFloorById(id: Long): FloorInfoVO

    fun getFloorList(searchCondition: String, page: Int, size: Int): List<FloorListVO>

    fun getFloorEntityById(id: Long): FloorEntity
}
